### PR TITLE
make HTTPMethod.hasRequestBody internal

### LINKED ISSUE #151: HTTPMethod.hasRequestBody `.unlikely`
The current implementation of `hasRequestBody` uses `unlikely` for a lot of cases where a body is actually `very likely` or a plain `.yes` (not sure what the different is, a PUT can also have an empty body).

Examples MKCOL, PROPFIND, REPORT, MKCALENDAR, etc.

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

HTTPMethod.hasRequestBody was unnecessarily public.

Modifications:

make it internal

Result:

fixes #151

### CODE REVIEW

File: docs/public-api-changes-NIO1-to-NIO2.md [hunk 1]
```diff
@@ -86,3 +86,5 @@
 - change  `ChannelPipeline.addHandler(_:after:)` to  `ChannelPipeline.addHandler(_:postion:)` where `position` can be `.first`, `.last`, `.before(ChannelHandler)`, and `.after(ChannelHandler)`
 - Change `HTTPServerProtocolUpgrader` `protocol` to require `buildUpgradeResponse` to take a `channel` and return an `EventLoopFuture<HTTPHeaders>`.
 - `EmbeddedChannel.writeInbound/Outbound` are now `throwing`
+- `HTTPMethod.hasRequestBody` and the `HTTPMethod.HasBody` type have been removed from the public API
+- `CircularBuffer.Index` is no longer `Strideable`
```
Comment (normanmaurer) [inline docs/public-api-changes-NIO1-to-NIO2.md]: This dos not seem to belong to this PR ?

Comment (weissi) [reply docs/public-api-changes-NIO1-to-NIO2.md]: @normanmaurer yeah, I sneaked that one in ;)
